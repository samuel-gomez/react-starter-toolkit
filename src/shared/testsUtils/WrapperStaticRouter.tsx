/* eslint-disable react/display-name */
import { ReactNode } from 'react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

const WrapperStaticRouter =
  (route = '/') =>
  ({ children }: { children: ReactNode }) =>
    <MemoryRouterProvider url={route}>{children}</MemoryRouterProvider>;

export default WrapperStaticRouter;
