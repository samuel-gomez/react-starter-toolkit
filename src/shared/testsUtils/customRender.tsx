/* eslint-disable react/display-name */
import { render, RenderOptions } from '@testing-library/react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

import FetchProvider from 'providers/FetchProvider';
import UserProvider from 'providers/UserProvider';
import { ReactElement, ReactNode } from 'react';
import { MOCK_API_URL } from '.';

type TMockProvider = {
  [x: string]: Record<string, unknown | number | string> | string;
};

const MockProviders =
  ({ role = '', ...testMock }: TMockProvider) =>
  ({ children }: { children: ReactNode }) => {
    const useOidcAccessTokenFn = jest.fn().mockReturnValue({ accessToken: 'accessToken' });
    const useOidcUserFn = jest.fn().mockReturnValueOnce({ oidcUser: { member_of: [`CN=${role}`], name: 'Samuel Gomez' } });
    return (
      <UserProvider useOidcUserFn={useOidcUserFn}>
        <FetchProvider
          apiUrl={MOCK_API_URL}
          fetchConfig={{
            headers: { testMock: JSON.stringify(testMock) },
          }}
          useOidcAccessTokenFn={useOidcAccessTokenFn}
        >
          <MemoryRouterProvider url="">{children}</MemoryRouterProvider>
        </FetchProvider>
      </UserProvider>
    );
  };

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>, testMock: TMockProvider = { role: '' }) =>
  render(ui, { wrapper: MockProviders(testMock), ...options });

// re-export everything
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// override render method
export { customRender as render };
