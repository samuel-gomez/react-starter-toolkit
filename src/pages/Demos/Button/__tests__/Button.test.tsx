import { renderWithWrapperStaticRouter } from 'shared/testsUtils';
import ButtonPage, { code } from '../Button';

describe('<ButtonPage/>', () => {
  it('Should render ButtonPage', () => {
    const { asFragment } = renderWithWrapperStaticRouter(<ButtonPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});

const cleanString = (str: string) => str.replace(/\s/g, '').replace(/\n/g, '').replace(/\t/g, '').trim();

describe('code', () => {
  const onChange = jest.fn();
  const label = 'valider';
  const className = 'af-btn';
  const classModifier = 'af-btn';
  const icon = 'ok';
  const disabled = true;

  it('Should render ButtonPage with icon', () => {
    const result = code({ label, className, classModifier, icon, disabled, onChange });
    expect(cleanString(result)).toEqual(
      cleanString(`<Button disabled={${disabled}} className="${className}" classModifier="${classModifier}" type="submit" onClick={onClick} >
          <i className="glyphicon glyphicon-${icon}"></i><span className="af-btn__text">${label}</span>
        </Button>
    `),
    );
  });

  it('Should render ButtonPage without icon', () => {
    const result = code({ label, className, classModifier, disabled, onChange });
    expect(cleanString(result)).toEqual(
      cleanString(`<Button disabled={${disabled}} className="${className}" classModifier="${classModifier}" type="submit" onClick={onClick} >
          <span className="af-btn__text">${label}</span>
        </Button>
    `),
    );
  });
});
