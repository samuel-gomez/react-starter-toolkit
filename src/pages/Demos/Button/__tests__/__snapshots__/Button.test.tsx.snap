// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPage/> Should render ButtonPage 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="Toolkit React Starter"
              class="af-logo__brand"
              src="slash-logo.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            Toolkit React Starter
            <span
              class="af-header__subtitle"
            >
              by Slash Design System
            </span>
          </h2>
        </div>
        <div
          class="af-info-user"
        >
          <a
            class="af-info-user__link"
            href="#"
            title="Voir mon profil"
          >
            <span
              class="af-info-user__name"
            />
          </a>
        </div>
      </header>
    </div>
  </div>
  <div
    class="af-nav-container"
  >
    <button
      aria-controls="mainmenu"
      aria-haspopup="true"
      class="mask fade "
      id="togglemenu"
      type="button"
    />
    <nav
      aria-label="Menu principal"
      class="af-nav af-drawer left "
      role="navigation"
    >
      <div
        class="container af-nav__wrapper"
      >
        <div
          class="af-nav__aside-header"
        >
          <h4
            class="af-nav__aside-title"
          >
            Menu
          </h4>
          <button
            aria-controls="mainmenu"
            aria-haspopup="true"
            class="af-nav__aside-close"
            id="togglemenu"
            type="button"
          >
            ×
          </button>
        </div>
        <ul
          aria-labelledby="togglemenu"
          class="af-nav__list "
          id="mainmenu"
          role="menubar"
        >
          <li
            class="af-nav__item af-nav__item--active "
            role="none"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              aria-label="Accueil"
              class="af-nav__link"
              href="/"
              role="menuitem"
              tabindex="0"
            >
              Accueil
            </a>
          </li>
          <li
            class="af-nav__item af-nav__item--haschild "
            role="none"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              aria-label="Démos"
              class="af-nav__link"
              href="/demos"
              role="menuitem"
              tabindex="-1"
            >
              Démos
            </a>
            <i
              class="glyphicon glyphicon-arrow-xs-down"
            />
            <ul
              aria-label=""
              class="af-nav__list af-nav__list--children"
              role="menu"
            >
              <li
                class="af-nav__item af-nav__item--separator "
                role="none"
              >
                <span
                  aria-expanded="false"
                  class="af-nav__link"
                  role="menuitem"
                  tabindex="-1"
                >
                  Pages
                </span>
              </li>
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Membres"
                  class="af-nav__link"
                  href="/demos/members"
                  role="menuitem"
                  tabindex="-1"
                >
                  Membres
                </a>
              </li>
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Rechercher"
                  class="af-nav__link"
                  href="/demos/search-members"
                  role="menuitem"
                  tabindex="-1"
                >
                  Rechercher
                </a>
              </li>
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Modal"
                  class="af-nav__link"
                  href="/demos/modal"
                  role="menuitem"
                  tabindex="-1"
                >
                  Modal
                </a>
              </li>
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Notification"
                  class="af-nav__link"
                  href="/demos/notification"
                  role="menuitem"
                  tabindex="-1"
                >
                  Notification
                </a>
              </li>
              <li
                class="af-nav__item af-nav__item--separator "
                role="none"
              >
                <span
                  aria-expanded="false"
                  class="af-nav__link"
                  role="menuitem"
                  tabindex="-1"
                >
                  Composants Toolkit
                </span>
              </li>
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Button"
                  class="af-nav__link"
                  href="/demos/button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Button
                </a>
              </li>
            </ul>
          </li>
          <li
            class="af-nav__item af-nav__item--haschild "
            role="none"
          >
            <span
              aria-expanded="false"
              class="af-nav__link"
              role="menuitem"
              tabindex="-1"
            >
              Pages
            </span>
            <i
              class="glyphicon glyphicon-arrow-xs-down"
            />
            <ul
              aria-label=""
              class="af-nav__list af-nav__list--children"
              role="menu"
            >
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Not found"
                  class="af-nav__link"
                  href="/notfound"
                  role="menuitem"
                  tabindex="-1"
                >
                  Not found
                </a>
              </li>
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Forbidden"
                  class="af-nav__link"
                  href="/forbidden"
                  role="menuitem"
                  tabindex="-1"
                >
                  Forbidden
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <a
    aria-controls="mainmenu"
    aria-haspopup="true"
    class="btn af-title-bar__mobile-menu af-btn--circle"
    href="#"
    id="togglemenu"
    role="button"
    tabindex="0"
    title="Toggle menu"
  >
    <i
      class="glyphicon glyphicon-menu-hamburger"
    />
  </a>
  <div
    class="af-title-bar"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <h1
        class="af-title-bar__title"
      >
        Demo button
      </h1>
    </div>
  </div>
  <main
    class="af-main af-main--demo"
  >
    <section
      class="container"
    >
      <h1
        class="af-title--content"
      >
        Examples
      </h1>
      <section
        class="af-editor"
      >
        <article
          class="af-livecode af-livecode--with-editor"
        >
          <div
            style="background: white; padding: 2rem; place-self: center;"
          >
            <button
              class="af-btn"
              type="submit"
            >
              <span
                class="af-btn__text"
              >
                valider
              </span>
            </button>
          </div>
          <pre
            class="prism-code language-jsx"
            style="color: rgb(248, 248, 242); background-color: rgb(40, 42, 54); font-size: 0.8rem;"
          >
            <div
              class="token-line"
              style="color: rgb(248, 248, 242);"
            >
              <span
                class="token plain"
                style="display: inline-block;"
              >
                

              </span>
            </div>
            <div
              class="token-line"
              style="color: rgb(248, 248, 242);"
            >
              <span
                class="token plain"
              >
                  
              </span>
              <span
                class="token tag punctuation"
                style="color: rgb(248, 248, 242);"
              >
                &lt;
              </span>
              <span
                class="token tag class-name"
                style="color: rgb(255, 121, 198);"
              >
                Button
              </span>
              <span
                class="token tag"
                style="color: rgb(255, 121, 198);"
              >
                 
              </span>
              <span
                class="token tag attr-name"
                style="color: rgb(241, 250, 140);"
              >
                disabled
              </span>
              <span
                class="token tag script language-javascript script-punctuation punctuation"
                style="color: rgb(248, 248, 242);"
              >
                =
              </span>
              <span
                class="token tag script language-javascript punctuation"
                style="color: rgb(248, 248, 242);"
              >
                {
              </span>
              <span
                class="token tag script language-javascript boolean"
                style="color: rgb(255, 121, 198);"
              >
                false
              </span>
              <span
                class="token tag script language-javascript punctuation"
                style="color: rgb(248, 248, 242);"
              >
                }
              </span>
              <span
                class="token tag"
                style="color: rgb(255, 121, 198);"
              >
                 
              </span>
              <span
                class="token tag attr-name"
                style="color: rgb(241, 250, 140);"
              >
                className
              </span>
              <span
                class="token tag attr-value punctuation attr-equals"
                style="color: rgb(248, 248, 242);"
              >
                =
              </span>
              <span
                class="token tag attr-value punctuation"
                style="color: rgb(248, 248, 242);"
              >
                "
              </span>
              <span
                class="token tag attr-value"
                style="color: rgb(255, 121, 198);"
              >
                af-btn
              </span>
              <span
                class="token tag attr-value punctuation"
                style="color: rgb(248, 248, 242);"
              >
                "
              </span>
              <span
                class="token tag"
                style="color: rgb(255, 121, 198);"
              >
                 
              </span>
              <span
                class="token tag attr-name"
                style="color: rgb(241, 250, 140);"
              >
                classModifier
              </span>
              <span
                class="token tag attr-value punctuation attr-equals"
                style="color: rgb(248, 248, 242);"
              >
                =
              </span>
              <span
                class="token tag attr-value punctuation"
                style="color: rgb(248, 248, 242);"
              >
                "
              </span>
              <span
                class="token tag attr-value punctuation"
                style="color: rgb(248, 248, 242);"
              >
                "
              </span>
              <span
                class="token tag"
                style="color: rgb(255, 121, 198);"
              >
                 
              </span>
              <span
                class="token tag attr-name"
                style="color: rgb(241, 250, 140);"
              >
                type
              </span>
              <span
                class="token tag attr-value punctuation attr-equals"
                style="color: rgb(248, 248, 242);"
              >
                =
              </span>
              <span
                class="token tag attr-value punctuation"
                style="color: rgb(248, 248, 242);"
              >
                "
              </span>
              <span
                class="token tag attr-value"
                style="color: rgb(255, 121, 198);"
              >
                submit
              </span>
              <span
                class="token tag attr-value punctuation"
                style="color: rgb(248, 248, 242);"
              >
                "
              </span>
              <span
                class="token tag"
                style="color: rgb(255, 121, 198);"
              >
                 
              </span>
              <span
                class="token tag attr-name"
                style="color: rgb(241, 250, 140);"
              >
                onClick
              </span>
              <span
                class="token tag script language-javascript script-punctuation punctuation"
                style="color: rgb(248, 248, 242);"
              >
                =
              </span>
              <span
                class="token tag script language-javascript punctuation"
                style="color: rgb(248, 248, 242);"
              >
                {
              </span>
              <span
                class="token tag script language-javascript"
                style="color: rgb(255, 121, 198);"
              >
                onClick
              </span>
              <span
                class="token tag script language-javascript punctuation"
                style="color: rgb(248, 248, 242);"
              >
                }
              </span>
              <span
                class="token tag"
                style="color: rgb(255, 121, 198);"
              >
                 
              </span>
              <span
                class="token tag punctuation"
                style="color: rgb(248, 248, 242);"
              >
                &gt;
              </span>
              <span
                class="token plain-text"
              />
            </div>
            <div
              class="token-line"
              style="color: rgb(248, 248, 242);"
            >
              <span
                class="token plain-text"
              >
                    
              </span>
              <span
                class="token tag punctuation"
                style="color: rgb(248, 248, 242);"
              >
                &lt;
              </span>
              <span
                class="token tag"
                style="color: rgb(255, 121, 198);"
              >
                span
              </span>
              <span
                class="token tag"
                style="color: rgb(255, 121, 198);"
              >
                 
              </span>
              <span
                class="token tag attr-name"
                style="color: rgb(241, 250, 140);"
              >
                className
              </span>
              <span
                class="token tag attr-value punctuation attr-equals"
                style="color: rgb(248, 248, 242);"
              >
                =
              </span>
              <span
                class="token tag attr-value punctuation"
                style="color: rgb(248, 248, 242);"
              >
                "
              </span>
              <span
                class="token tag attr-value"
                style="color: rgb(255, 121, 198);"
              >
                af-btn__text
              </span>
              <span
                class="token tag attr-value punctuation"
                style="color: rgb(248, 248, 242);"
              >
                "
              </span>
              <span
                class="token tag punctuation"
                style="color: rgb(248, 248, 242);"
              >
                &gt;
              </span>
              <span
                class="token plain-text"
              >
                valider
              </span>
              <span
                class="token tag punctuation"
                style="color: rgb(248, 248, 242);"
              >
                &lt;/
              </span>
              <span
                class="token tag"
                style="color: rgb(255, 121, 198);"
              >
                span
              </span>
              <span
                class="token tag punctuation"
                style="color: rgb(248, 248, 242);"
              >
                &gt;
              </span>
              <span
                class="token plain-text"
              />
            </div>
            <div
              class="token-line"
              style="color: rgb(248, 248, 242);"
            >
              <span
                class="token plain-text"
              >
                  
              </span>
              <span
                class="token tag punctuation"
                style="color: rgb(248, 248, 242);"
              >
                &lt;/
              </span>
              <span
                class="token tag class-name"
                style="color: rgb(255, 121, 198);"
              >
                Button
              </span>
              <span
                class="token tag punctuation"
                style="color: rgb(248, 248, 242);"
              >
                &gt;
              </span>
              <span
                class="token plain"
              />
            </div>
            <div
              class="token-line"
              style="color: rgb(248, 248, 242);"
            >
              <span
                class="token plain"
                style="display: inline-block;"
              >
                

              </span>
            </div>
          </pre>
        </article>
        <form
          class="af-form-editor"
        >
          <div
            class="af-form-editor__field"
          >
            <label
              class="af-form__group-label"
              for="classModifier"
            >
              classModifier
            </label>
            <div
              class="af-form__select"
            >
              <div
                class="af-form__select-container"
              >
                <select
                  aria-label="select-classModifier"
                  class="af-form__input-select"
                  id="classModifier"
                  name="classModifier"
                >
                  <option
                    value=""
                  >
                    - Sélectionner -
                  </option>
                  <option
                    value="success"
                  >
                    Success
                  </option>
                  <option
                    value="danger"
                  >
                    Danger
                  </option>
                  <option
                    value="reverse"
                  >
                    Reverse
                  </option>
                  <option
                    value="disabled"
                  >
                    Disabled
                  </option>
                  <option
                    value="small"
                  >
                    Small
                  </option>
                  <option
                    value="hasiconLeft"
                  >
                    HasiconLeft
                  </option>
                  <option
                    value="hasiconRight"
                  >
                    HasiconRight
                  </option>
                  <option
                    value="success hasiconLeft"
                  >
                    Success & hasiconLeft
                  </option>
                  <option
                    value="danger hasiconLeft"
                  >
                    Danger & hasiconLeft
                  </option>
                  <option
                    value="reverse hasiconLeft"
                  >
                    Reverse & hasiconLeft
                  </option>
                  <option
                    value="disabled hasiconLeft"
                  >
                    Disabled & hasiconLeft
                  </option>
                  <option
                    value="success hasiconRight"
                  >
                    Success & hasiconRight
                  </option>
                  <option
                    value="danger hasiconRight"
                  >
                    Danger & hasiconRight
                  </option>
                  <option
                    value="reverse hasiconRight"
                  >
                    Reverse & hasiconRight
                  </option>
                  <option
                    value="disabled hasiconRight"
                  >
                    Disabled & hasiconRight
                  </option>
                </select>
                <span
                  aria-controls="classModifier"
                  class="glyphicon glyphicon-menu-down"
                />
              </div>
            </div>
          </div>
          <div
            class="af-form-editor__field"
          >
            <label
              class="af-form__group-label"
              for="className"
            >
              className
            </label>
            <input
              class="af-form__input-text"
              id="className"
              name="className"
              type="text"
              value="af-btn"
            />
          </div>
          <div
            class="af-form-editor__field"
          >
            <label
              class="af-form__group-label"
              for="label"
            >
              label
            </label>
            <input
              class="af-form__input-text"
              id="label"
              name="label"
              type="text"
              value="valider"
            />
          </div>
          <div
            class="af-form-editor__field"
          >
            <label
              class="af-form__group-label"
              for="disabled"
            >
              disabled
            </label>
            <div
              class="af-form__checkbox-toggle af-form__checkbox-toggle--disabled"
            >
              <input
                class="af-form__input-checkbox"
                disabled=""
                id="disabled"
                mode="toggle"
                name="disabled"
                type="checkbox"
              />
              <label
                class="af-form__label"
                for="disabled"
              >
                <span
                  class="af-form__indicator"
                >
                  <i
                    class="glyphicon glyphicon-ok"
                  />
                </span>
                <span
                  class="af-form__description"
                />
              </label>
            </div>
          </div>
          <div
            class="af-form-editor__field"
          >
            <label
              class="af-form__group-label"
              for="icon"
            >
              icon
            </label>
            <div
              class="af-form__select"
            >
              <div
                class="af-form__select-container"
              >
                <select
                  aria-label="select-icon"
                  class="af-form__input-select"
                  id="icon"
                  name="icon"
                >
                  <option
                    value=""
                  >
                    - Sélectionner -
                  </option>
                  <option
                    value="adjust"
                  >
                    adjust
                  </option>
                  <option
                    value="alert"
                  >
                    alert
                  </option>
                  <option
                    value="arrowthin-right"
                  >
                    arrowthin-right
                  </option>
                  <option
                    value="arrowthin-left"
                  >
                    arrowthin-left
                  </option>
                  <option
                    value="ban-circle"
                  >
                    ban-circle
                  </option>
                  <option
                    value="axa-switch"
                  >
                    axa-switch
                  </option>
                  <option
                    value="bookmark"
                  >
                    bookmark
                  </option>
                  <option
                    value="close"
                  >
                    close
                  </option>
                  <option
                    value="comment"
                  >
                    comment
                  </option>
                  <option
                    value="download"
                  >
                    download
                  </option>
                  <option
                    value="duplicate"
                  >
                    duplicate
                  </option>
                  <option
                    value="earphone"
                  >
                    earphone
                  </option>
                  <option
                    value="education"
                  >
                    education
                  </option>
                  <option
                    value="envelope"
                  >
                    envelope
                  </option>
                  <option
                    value="euro-symbol"
                  >
                    euro-symbol
                  </option>
                  <option
                    value="eye-open"
                  >
                    eye-open
                  </option>
                  <option
                    value="file"
                  >
                    file
                  </option>
                  <option
                    value="filter"
                  >
                    filter
                  </option>
                  <option
                    value="flag"
                  >
                    flag
                  </option>
                  <option
                    value="flash"
                  >
                    flash
                  </option>
                  <option
                    value="floppy-disk"
                  >
                    floppy-disk
                  </option>
                  <option
                    value="fullscreen"
                  >
                    fullscreen
                  </option>
                  <option
                    value="heart"
                  >
                    heart
                  </option>
                  <option
                    value="home"
                  >
                    home
                  </option>
                  <option
                    value="imprimer"
                  >
                    imprimer
                  </option>
                  <option
                    value="map-marker"
                  >
                    map-marker
                  </option>
                  <option
                    value="ok"
                  >
                    ok
                  </option>
                  <option
                    value="paperclip"
                  >
                    paperclip
                  </option>
                  <option
                    value="play"
                  >
                    play
                  </option>
                  <option
                    value="plus"
                  >
                    plus
                  </option>
                  <option
                    value="refresh"
                  >
                    refresh
                  </option>
                  <option
                    value="trash"
                  >
                    trash
                  </option>
                </select>
                <span
                  aria-controls="icon"
                  class="glyphicon glyphicon-menu-down"
                />
              </div>
            </div>
          </div>
        </form>
      </section>
    </section>
  </main>
  <footer
    class="af-footer"
  >
    <div
      class="container-fluid container"
    >
      <a
        class="af-logo"
        href="https://www.axa.fr/"
        target="blank"
        title="Site Axa"
      >
        <img
          alt="Logo Axa"
          class="af-logo__brand"
          src="logo-axa.svg"
        />
      </a>
      <div
        class="af-footer-content"
      >
        © 2022 AXA Tous droits réservés - v0.1.6
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
